src/test/groovy/org/gradlex/buildparameters/BuildParametersPluginFuncTest.groovy

```diff
@@ -243,14 +243,13 @@ class BuildParametersPluginFuncTest extends Specification {
                         defaultValue = "localhost"
                     }
                     string("port") {
-                        defaultValue = "5432"
```

Comment:
I changed the test here, because "optional" + "group" is related to this issue (we inject services to create providers). I think it is good to have this combination covered - just in case.